app/src/main/java/com/github/se/polyfit/viewmodel/post/ViewPostViewModel.kt

```diff
@@ -19,17 +27,43 @@ import kotlinx.coroutines.flow.StateFlow
 @HiltViewModel
 class ViewPostViewModel
 @Inject
-constructor(private val postFirebaseRepository: PostFirebaseRepository) : ViewModel() {
+constructor(
+    private val postFirebaseRepository: PostFirebaseRepository,
+    private val postLocalRepository: PostLocationModel
+) : ViewModel() {
   private val _posts: MutableStateFlow<List<Post>> = MutableStateFlow(mutableListOf())
+  val posts: StateFlow<List<Post>> = _posts
 
-  private val _isFetching: MutableStateFlow<Boolean> = MutableStateFlow(false)
-  val isFetching: StateFlow<Boolean> = _isFetching
+  private val _isFetching: MutableLiveData<Boolean> = MutableLiveData(false)
+  val isFetching: LiveData<Boolean> = _isFetching
+
+  private val _location = MutableLiveData<Location>()
+  val location: LiveData<Location> = _location
 
   init {
-    postFirebaseRepository.initializeFirebaseListener(viewModelScope)
+
+    viewModelScope.launch {
+      _isFetching.postValue(true)
+      _location.value =
+          postLocalRepository.getCurrentLocation(
+              CurrentLocationRequest.Builder().setPriority(Priority.PRIORITY_HIGH_ACCURACY).build())
+    }
+    _location.observeForever { getNearbyPosts(it) }
   }
 
-  fun getAllPost(): LiveData<List<Post>> {
-    return postFirebaseRepository.posts
+  fun getNearbyPosts(location: Location) {
+
+    viewModelScope.launch {
+      withContext(Dispatchers.Main) {
```

Comment:
I think that this double dispatcher is catching concurrency issues that sometimes cause the spinner to loop forever. I replaced it with this, and it seems to work:

```suggestion
viewModelScope.launch {
    withContext(Dispatchers.Main) {